File: arrays/java/Solution01.java
package arrays.java;

import java.util.Arrays;

class Solution01 {

    public int removeDuplicates(int[] nums) {

        if (nums.length == 0) {
            return 0;
        }

        int k = 0;

        Arrays.sort(nums);

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[k]) {
                k++;
                nums[k] = nums[i];
            }
        }
        return k + 1;
    }
}

File: arrays/go/solution01.go
package leetcode

import "sort"

func RemoveDuplicates(nums []int) int {

	if len(nums) == 0 {
		return 0
	}

	sort.Ints(nums)

	k := 0

	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[k] {
			k++
			nums[k] = nums[i]
		}
	}
	return k + 1
}
